bin/check-upgrade.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+error="0"
+
+
+if [ "$1" = "snmp" ]; then
+    if [ -z "$3" ]; then
+        param_port="161"
+    else
+        param_port=$3
+    fi
+
+    if [ -z "$4" ]; then
+        param_community="public"
+    else
+        param_community=$4
+    fi
+
+    if ! routeros_installed=$(snmpget -O qv -v 2c -c "$param_community" "$2":"$param_port" SNMPv2-SMI::enterprises.14988.1.1.4.4.0 2> /dev/null); then
+        error="Could not establish an SNMP connection to the device!"
+    else
+        routeros_installed=$(echo "$routeros_installed" | tr -d '"')
+    fi
+elif [ "$1" = "ssh" ]; then
+    if [ -z "$3" ]; then
+        param_port="22"
+    else
+        param_port=$3
+    fi
+
+    if [ -z "$4" ]; then
+        param_user="user"
+    else
+        param_user=$4
+    fi
+
+    if ! routeros_installed=$(ssh -q -p "$param_port" "$param_user"@"$2" ':put [/system package get system version]'); then
+        error="Could not establish an SSH connection to the device!"
+    else
+        routeros_installed=$(echo "$routeros_installed" | tr -d "\\r\\n")
+    fi
+else
+    echo "Use SNMP or SSH as connection type!"
+    exit 2
+fi
+
+
+if [ "$error" = "0" ]; then
+    # select the version
+    case $5 in
+        "stable")
+            param_version="6"
+            ;;
+        "release-candidate")
+            param_version="6rc"
+            ;;
+        "bugfix")
+            param_version="6fix"
+            ;;
+        *)
+            param_version="6"
+            ;;
+    esac
+
+    # check the MikroTik server for upgrades
+    if ! routeros_available=$(curl -fsA "check_routeros-upgrade" https://download.mikrotik.com/routeros/LATEST.$param_version); then
```

Comment:
just a thought on enhancing this to allow it to be an argument for the url as someone might need to host their own in airgapped setups. Not required but would be nice to have.

Reply:
You mean something like proxy vars?

Reply:
no I mean maybe I am required to not allow external traffic even through a proxy. For proxies you can use the ENV vars and curl will honor it. What I mean is that maybe I want to host the content on:  https://download.domain.tld/routeros/LATEST.$param_version

Reply:
I added an ENV var. Now you can replace the default server if you want.